### ISSUE TITLE
Integration tests for components under islands

### ISSUE THREAD

**Issue by nmashchenko** [streamdal/streamdal#520]
Hey guys! 

Wanted to check in as I wrapped up unit tests as planned originally and want to do some integration tests for `islands` folder and see if you would be interested in that. I can also do E2E and extend those you already have but since you are missing integration I wanted to make sure they can be useful first.

Let me know and excited to keep working & contributing to the repo!

**Comment by uzarubin** [comment]
Of course! The more tested the repo, the better. Island tests are especially needed.
